pydantic_settings/sources/utils.py

```diff
@@ -54,6 +54,10 @@ def _annotation_is_complex(annotation: type[Any] | None, metadata: list[Any]) ->
 
     origin = get_origin(annotation)
 
+    # Check if annotation is of the form Union[type, ...].
+    if typing_objects.is_union(origin):
+        return _union_is_complex(annotation, metadata)
```

Comment:
@hramezani From what I observed, it appears to be related to incorrectly determining that nested annotated unions are not complex. This change fixed it, but causes other tests fail. I'm not sure if it is the correct fix. Can you take a look?

Reply:
Sorry @kschwab for the delay.

I am also not sure what the correct fix is here. As the change breaks some existing tests, I would prefer to remove it.

Reply:
No worries @hramezani. I'm ok with closing this as a won't fix as well.